static/js/activity.js

```diff
@@ -329,6 +329,12 @@ function focus_gained() {
     }
 }
 
+function get_desktop_presence() {
+    if (window.electron_bridge !== undefined) {
+        window.electron_bridge.send_event('desktop_presence', DEFAULT_IDLE_TIMEOUT_MS / 1000);
```

Comment:
Hmm, why is the right model to pass the timeout to the desktop app here?

Reply:
This was done because the powerMonitor API in electron requires a timeout to detect state.

Reply:
Hmm, can someone explain precisely what that timeout is use for?  I would guess it's a polling/check frequency and thus a bound of the accuracy of information on the data from that API, not a timeout, in which case we might want to have a different value for that that's significantly shorter (e.g. 15 seconds).  

Also, while we're at it, does the electron power monitor API give us a notification when the user goes from idle => active, active => idle, or both?

Reply:
```
powerMonitor.querySystemIdleState(idleThreshold, callback)
```
Here the idleThreshold is the time after which the user will be considered idle (no keyboard/mouse movement). We need call this function after very t seconds (which you are referring to as polling frequency). 
So the overall flow will look something like this-
```
setTimeout(powerMonitor.querySystemIdleState(idleThreshold, idleState=> {
        if (idleState==='idle'){
                // logic to set user as idle in the webapp
        }
}, pollingFrequency);
```

For example- 
If we implement something like this in the main process-
```
setInterval(() => {
	electron.powerMonitor.querySystemIdleState(20, idleState => {
		console.log(idleState);
	});
}, 1000);
```
Here, the state will be checked every 1000 ms (1s), and if the user has been idle for the last 20s, 'idle' will be logged to console after every 1s, until user makes an activity.

Hope that clears things up.